From 4b83de426c6f485016fb95bc340d2bdabeafea67 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sun, 20 Nov 2022 18:12:58 +0100
Subject: [PATCH 6/9] Fix a lot of Huawei EMUI9 denied under android 13

Add support off splash2 Huawei logging
Authorize a lot of process to read properties
Fix various denied
---
 sepolicy/huawei.te | 115 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 115 insertions(+)

diff --git a/sepolicy/huawei.te b/sepolicy/huawei.te
index 2988ed6..6e15e7a 100644
--- a/sepolicy/huawei.te
+++ b/sepolicy/huawei.te
@@ -35,3 +35,118 @@ allowxperm vendor_init { teecd_data_file }:dir ioctl {
 type sys_block_mmcblk0, file_type;
 allow init sys_block_mmcblk0:lnk_file read;
 allow init sysfs_zram:lnk_file read;
+
+
+# --------------------- Not need to boot but this rules are denied  ----------------------------------------
+
+# Add support off splash2 Huawei logging
+type splash2_data_file, file_type;
+
+allow splash2_data_file self:filesystem { associate };
+allow init splash2_data_file:filesystem { mount relabelto relabelfrom };
+allow init splash2_data_file:dir { getattr mounton };
+allow vendor_init splash2_data_file:dir { open read setattr getattr };
+allow vendor_init splash2_data_file:file { open write create setattr getattr };
+allow vendor_init splash2_data_file:filesystem { getattr };
+allow installd splash2_data_file:filesystem { quotaget };
+allow rild splash2_data_file:dir { open read open };
+allow rild splash2_data_file:file { open read write getattr };
+
+
+
+# Fix all SEdenied 
+
+# allow vold modem_secure_file:filesystem { unmount };
+# allow hwemerffu_service proc:file { open read };
+# allow netutils_wrapper hinetmanager:fd { use };
+# allow hinetmanager self:capability { dac_override };
+
+allow vold sys_block_mmcblk0:file { write };
+allow vold sysfs_zram:file { write };
+
+allow kernel device:dir { add_name write };
+allow kernel self:capability { mknod };
+allow kernel device:chr_file { create setattr };
+
+type teecd_data_file_system, data_file_type;
+allow vendor_init device:chr_file { getattr };
+allow vendor_init teecd_data_file_system:dir { relabelto getattr };
+allow vendor_init system_data_file:dir { add_name write setattr };
+
+allow init teecd_data_file:dir { mounton };
+allow init teecd_data_file:filesystem { mount relabelto relabelfrom };
+allow init sys_block_mmcblk0:file { open write };
+
+allow fsck block_device:blk_file { read open write ioctl };
+allow teecd_data_file self:filesystem { associate };
+allow rootfs labeledfs:filesystem { associate };
+allow tee self:capability { sys_ptrace };
+
+allow kernel device:chr_file { open write };
+allow oeminfo_nvm block_device:blk_file { ioctl getattr open read write };
+allow gpsdaemon system_data_file:lnk_file { read };
+allow oeminfo_nvm device:chr_file { ioctl open read write };
+allow rild system_data_file:lnk_file { read };
+allow kernel system_data_root_file:dir { open read setattr };
+allow kernel system_data_root_file:file { open setattr getattr read append };
+allow kernel unlabeled:file { read open };
+
+allow hi110x_daemon rootfs:dir { open read };
+allow hi110x_daemon self:fifo_file { ioctl };
+
+allow hal_camera_default system_data_file:lnk_file { read };
+allow surfaceflinger bootanim:dir { search };
+allow surfaceflinger bootanim:file { open read };
+allow hal_audio_default audioserver:fifo_file { write };
+allow installd self:capability { sys_ptrace };
+allow installd system_server:file { open read };
+allow installd system_server:dir { search };
+
+allow installd teecd_data_file:filesystem { quotaget };
+allow system_server sysfs_zram:lnk_file { read };
+allow system_server vendor_file:file { getattr open read };
+allow kernel sysfs_devices_system_cpu:file { write };
+allow network_stack fs_bpf:file { read write };
+
+# OTPERASE
+allowxperm oeminfo_nvm device:blk_file ioctl {
+  0x4d19
+};
+
+allow shell self:netlink_kobject_uevent_socket { create setopt getopt bind read };
+allow init self:netlink_kobject_uevent_socket { create setopt getopt bind read };
+
+
+
+# Authorize a lot of process to read default, config and system prop
+type hal_health_default, domain;
+type hal_camera_default, domain;
+type hal_audio_default, domain;
+type hal_keymaster_default, domain;
+
+get_prop(rild, config_prop)
+get_prop(rild, system_prop)
+get_prop(hal_health_default, default_prop)
+get_prop(hal_health_default, config_prop)
+get_prop(hal_fingerprint_default, default_prop)
+get_prop(hal_camera_default, default_prop)
+get_prop(hal_camera_default, config_prop)
+get_prop(hal_audio_default, default_prop)
+get_prop(hal_audio_default, config_prop)
+get_prop(hal_audio_default, system_prop)
+get_prop(hal_keymaster_default, config_prop)
+
+type hi110x_daemon, domain;
+type wpa_hisi, domain;
+type aptouch_daemon, domain;
+type gpsdaemon, domain;
+type atcmdserver, domain;
+
+get_prop(hi110x_daemon, default_prop)
+get_prop(wpa_hisi, default_prop)
+get_prop(aptouch_daemon, system_prop)
+get_prop(gpsdaemon, default_prop)
+
+get_prop(oeminfo_nvm, default_prop)
+get_prop(atcmdserver, default_prop)
+
-- 
2.25.1

